clash-prelude/src/Clash/Class/Counter/Internal.hs

```diff
@@ -80,6 +85,56 @@ instance KnownNat n => Counter (Unsigned n)
 instance KnownNat n => Counter (Signed n)
 instance KnownNat n => Counter (BitVector n)
 
+-- | @since 1.8.2
+instance Counter Bool
+-- | @since 1.8.2
+instance Counter Bit
+-- | @since 1.8.2
+instance Counter Int
+-- | @since 1.8.2
+instance Counter Int8
+-- | @since 1.8.2
+instance Counter Int16
+-- | @since 1.8.2
+instance Counter Int32
+-- | @since 1.8.2
+instance Counter Int64
+-- | @since 1.8.2
+instance Counter Word
+-- | @since 1.8.2
+instance Counter Word8
+-- | @since 1.8.2
+instance Counter Word16
+-- | @since 1.8.2
+instance Counter Word32
+-- | @since 1.8.2
+instance Counter Word64
+
+-- | @since 1.8.2
+deriving newtype instance Counter a => Counter (Identity a)
+
+-- | Opiniated instance for 'Maybe'. 'Nothing' is considered the minimum value,
+-- while @'Just' 'countMax'@ is considered the maximum value.
+--
```

Comment:
Doesn't seem too opinionated to me, doesn't Nothing being the minimum follow from the Ord instance of Maybe?

```
ghci> Nothing < Just ()
True
ghci> Just () < Nothing
False

```

Reply:
Ooh, you're right! I'll amend the PR.

Reply:
Thanks, Leon, that was what was niggling me, I remembered we discussed this but I couldn't recall what it was! :-D

Reply:
I guess it's fair as this class links `Ord` and `Bounded` which are normally explicitly unrelated.